### ISSUE TITLE
dnu diagnostics (error CS7038:failed to emit module)

### ISSUE THREAD

**Issue by seesharper** [dotnet/roslyn#7143]
I have a project that I try to build using "dnu build" and it fails with "CS7038:failed to emit module".
Is there any way I can turn on detailed diagnostics that will tell me more about the cause of the error?

"dnu --verbose build" does not provide anything useful

**Comment by leppie** [comment]
It is normally due to invalid code that is not detected properly by Roslyn. Search for CS7038 https://github.com/dotnet/roslyn/search?q=CS7038&type=Issues&utf8=%E2%9C%93 to see if you have anything similar in your code. If not. Try create a minimal repro.

**Comment by seesharper** [comment]
It compiles just fine with target framework set to dnx451, but fails under dnxcore50. 
So there is no way to get a detailed log from Roslyn that could give me a little more information about 
what the cause of the error might be? Otherwise this would be sort of finding the needle in a haystack.

**Comment by seesharper** [comment]
Another thing I discovered is that it happens only in release builds

**Comment by leppie** [comment]
Can you maybe run peverify on your debug build? Dont think it will help, but it might.

**Comment by seesharper** [comment]
Just tried it. peverify reports no errors on the debug build

**Comment by seesharper** [comment]
I am going to try to hunt this down, but Roslyn should seriously consider to provide more information in the event of failing to compile an assembly.

**Comment by leppie** [comment]
If you can, please let us know. I was attempting to fix another CS7038 bug over the weekend, but still havent been able to track down where the problem happens.

**Comment by seesharper** [comment]
I managed to narrow it down to this.

```
using System.Reflection;
public class Foo
{	   	
	public void SomeMethod()
	{
		const TypeAttributes TypeAttributes = TypeAttributes.Public;						
	}	
}
```

The error is caused by naming the variable exactly the same as the type name;

Changing the code to this makes it compile without the CS7038 error message.

```
using System.Reflection;
public class Foo
{	   	
	public void SomeMethod()
	{
		const TypeAttributes typeAttributes = TypeAttributes.Public;						
	}	
}
```

I did try this with an Enum defined in the same assembly, but then I was not able to reproduce the error. I don't even know if it has to be an enum or just a type from another assembly, but at least you have a minimal example that produces the error.

**Comment by leppie** [comment]
:+1:

**Comment by leppie** [comment]
I cant seem to reproduce it just from that code. Can you show what else is being done in the method?

**Comment by seesharper** [comment]
There is nothing more than this, but here is the project.json file.

```
{
  "version": "3.0.0.10",
  "dependencies": {
    
  },
  "frameworks": {   
    "dnxcore50": {
      "dependencies": {        
         "System.Collections": "4.0.10",
        "System.Collections.Concurrent": "4.0.10",
        "System.IO": "4.0.10",
        "System.Linq": "4.0.0",
        "System.Linq.Expressions": "4.0.10",
        "System.Reflection": "4.0.10",
        "System.Reflection.Emit": "4.0.0",
        "System.Reflection.Emit.Lightweight": "4.0.0",
        "System.Runtime": "4.0.20",
        "System.Runtime.Extensions": "4.0.10",
        "System.Text.Encoding": "4.0.10",
        "System.Text.RegularExpressions": "4.0.10",
        "System.Threading": "4.0.10",
        "System.Diagnostics.Tools": "4.0.0"        
      }
    }
  }
}
```

**Comment by gafter** [comment]
@seesharper Can you please package up the whole solution of your reduced repro along with a script that shows how you are building, into a zip file for us to look at?

**Comment by seesharper** [comment]
https://github.com/seesharper/roslyn_issue7143

Best regards
Bernhard Richter

**Comment by jaredpar** [comment]
Sorry this bug has gotten stale.  There were some early issues with DNU in this area.  The newer CLI builds though (successor to DNU) do not have this issue though.  Closing for now.  Please let us know if you still see this on CLI.
